### ISSUE TITLE
unable to update xtext-reference-project wizard examples to orbit 2024-06 / eclipse 2024-03

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#2962]
https://github.com/xtext/xtext-reference-projects/actions/runs/8408063499/job/23023767481

i am running out of time investigating whats going on ..,

am also not sure which java versions are visible to the eclipse in the tests.
locally i see java 11 disturbing

maybe we should say the java 11 users a goodbye when it comes to examples

**Comment by LorenzoBettini** [comment]
I see a few failures resolving update sites.. maybe it's just a network failure?

What do you mean you see java 11?

**Comment by LorenzoBettini** [comment]
That will depend on what they find in your os path..

**Comment by cdietrich** [comment]
@lorenzobettini
problem is persistent. builds not working.
https://github.com/xtext/xtext-reference-projects/actions/runs/8431370703

**Comment by LorenzoBettini** [comment]
These lines:

```
!MESSAGE bundle org.apache.felix.scr:2.2.10 (30) Circular reference detected trying to get service 
...
java.lang.Exception: stack trace
	at org.apache.felix.scr.impl.ComponentRegistry.enterCreate(ComponentRegistry.java:493)
	at org.apache.felix.scr.impl.BundleComponentActivator.enterCreate(BundleComponentActivator.java:722)
```

can also be found on other bug reports.

I don't think it's our fault... it's either m2e or platform...

This happens only after upgrading to the new Orbit, doesn't it?

**Comment by cdietrich** [comment]
Did not work neither this is why I started bumping all kind of stuff

**Comment by LorenzoBettini** [comment]
So this is the project that hasn't worked for a long time?

**Comment by LorenzoBettini** [comment]
OK, but concerning the one that is failing now: when has it started to fail?

**Comment by cdietrich** [comment]
the wizard gradle one? or the examples one?

**Comment by LorenzoBettini** [comment]
The one that you mention in this issue

**Comment by cdietrich** [comment]
the examples test is green on master
so it is related to the changes in my pr

**Comment by LorenzoBettini** [comment]
OK, so the failure is due to the new Orbit. That was my original question (see above).

**Comment by cdietrich** [comment]
indeed it looks like java 11 is visible "somehow" inside the tests

![homeAutomationExample(org eclipse xtext swtbot tests XtextExamplesTest)](https://github.com/eclipse/xtext/assets/238736/9fd8ad59-442f-49c2-885d-26acef03b907)

**Comment by cdietrich** [comment]
![domainModelExample(org eclipse xtext swtbot tests XtextExamplesTest)](https://github.com/eclipse/xtext/assets/238736/12f908fd-edd7-438d-9534-f5c283dd02d5)

**Comment by cdietrich** [comment]
branch with reduced set
https://github.com/xtext/xtext-reference-projects/tree/cd_tralala

**Comment by LorenzoBettini** [comment]
The one shown in the screenshot says "temurin", so it is the one installed with the setup java action... but we install Java 17, don't we?

Listing the contents of that system directory in a pre step would be helpful?

Is the project the test wants to build configured for Java 11 or Java 17? Lately, I saw Eclipse having a predefined option to discover installed Java versions and automatically configure them (something I immediately uncheck). If that happens, and the project is configured for Java 11 the running Eclipse will automatically use the discovered matching version.

Just a few thoughts

**Comment by cdietrich** [comment]
see the branch mentioned. all javas are there

**Comment by LorenzoBettini** [comment]
So if all javas are there, it might be due to what I said

**Comment by LorenzoBettini** [comment]
But what should I see in the mentioned branch? I was suggesting to print the contents of the java directory.

**Comment by cdietrich** [comment]
????
<img width="1355" alt="grafik" src="https://github.com/eclipse/xtext/assets/238736/a6846409-74e3-4574-b29a-1f118fbd2ab1">

**Comment by LorenzoBettini** [comment]
OK, then I think it happens as I said before (Eclipse automatically detects all the available JDKs and configures the JREs accordingly). Maybe we should configure those projects to use Java 17 if that's not already done. In that case, Java 17 should be used in our SWTBot tests. If that's already done, then the problem is somewhere else. Currently, I can't test it locally. Maybe in the next few days.

**Comment by cdietrich** [comment]
merged https://github.com/xtext/xtext-reference-projects/pull/413 
with that one we are not testing latest eclipse
